test/wg-easy/1.0.0/templates/deployment.yaml

```diff
@@ -0,0 +1,114 @@
+              mountPath: {{ $hostPathConfiguration.mountPath }}
+          {{ end }}
+          ports:
+            - name: udp
+              containerPort: 51820
+              protocol: UDP
+            - name: web
+              containerPort: 51821
+          env:
+            {{ $wgeasy := .Values.wgeasy }}
+            {{ $envList := (default list .Values.environmentVariables) }}
+            {{ $envList = mustAppend $envList (dict "name" "WG_HOST" "value" $wgeasy.host) }}
+            {{ $envList = mustAppend $envList (dict "name" "PASSWORD" "value" $wgeasy.password) }}
+            {{ $envList = mustAppend $envList (dict "name" "WG_PORT" "value" .Values.wgUDPPort) }}
+            {{ $envList = mustAppend $envList (dict "name" "WG_PERSISTENT_KEEPALIVE" "value" $wgeasy.keep_alive) }}
+            {{ $envList = mustAppend $envList (dict "name" "WG_MTU" "value" $wgeasy.client_mtu) }}
+            {{ $envList = mustAppend $envList (dict "name" "WG_DEFAULT_ADDRESS" "value" $wgeasy.client_address_range) }}
+            {{ $envList = mustAppend $envList (dict "name" "WG_DEFAULT_DNS" "value" $wgeasy.client_dns_server) }}
+            {{ if $wgeasy.allowed_ips }}
+            {{ $envList = mustAppend $envList (dict "name" "WG_ALLOWED_IPS" "value" (join "," $wgeasy.allowed_ips)) }}
+            {{ else }}
+            {{ $envList = mustAppend $envList (dict "name" "WG_ALLOWED_IPS" "value" ("0.0.0.0/0,::/0")) }}
+            {{ end }}
+            {{ include "common.containers.environmentVariables" (dict "environmentVariables" $envList) | nindent 12 }}
+          readinessProbe:
+            httpGet:
+              path: /
+              port: 51821
+            failureThreshold: 5
+            periodSeconds: 15
+          livenessProbe:
+            httpGet:
+              path: /
+              port: 51821
+            failureThreshold: 5
+            periodSeconds: 15
+          startupProbe:
+            httpGet:
+              path: /
+              port: 51821
+            initialDelaySeconds: 5
+            failureThreshold: 40
+            periodSeconds: 15
+          {{ $ip := .Values.wgeasy.client_address_range | replace "x" "0" }}
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                  - /bin/bash
+                  - -c
+                  - |
+                    echo "Deleting routes created by the app..."
+                    netmask=$(ip route | grep {{ $ip }})
+                    netmask=$(echo $netmask | grep -o -E '/.\d*')
+                    netmask=${netmask#/}
+                    echo "Matched routes to delete... {{ $ip }}/$netmask"
+                    # Don't try to delete routes if steps above didn't grep-ed anything
+                    if [ ! "$netmask" == "" ]; then
+                      ip route del {{ $ip }}/$netmask || echo "Route deletion failed..."
+                    fi
+                    echo "Routes deleted..."
```

Comment:
We should be deleting the wireguard interface as well

Reply:
Also what would happen if 2 instances of the same app are installed ? Do we have a conflict over interface name ?

Reply:
2 Instances with on hostNet won't work.
As the GUI port is not configurable. and it's port will conlict with the existing instance.


On app removal wg0 is removed, will check now if it also gets removed on scale down

Reply:
Modified it a bit. App can use different port for both the WG port and webUI.
But the problem is that the `wg0` interface name is hardcoded, and I don't see a way to modify that.
So while technically you can run 2 instances (or more). It's a limitation from the app that it's using a hardcoded interface name
---

Extended the script a bit to also clear the interface on scale down and/or deletion.